### PR TITLE
adding an http server for a rest api

### PR DESCRIPTION
**Pull request by luthermonson**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -40,6 +34,19 @@ func main() {
 			Name:   "debug",
 			EnvVar: "DEBUG",
 		},
+		cli.BoolFlag{
+			Name:   "api",
```
Comment (Oats87) [inline main.go]: Shouldn't these be something like `enable-api` to be more descriptive, as it's a flag not a "command"?

File: main.go [hunk 2]
```diff
@@ -40,6 +34,19 @@ func main() {
...
+			EnvVar: "API",
+		},
+		cli.StringFlag{
+			Name:   "api-address",
+			EnvVar: "API_ADDRESS",
+			Value:  "0.0.0.0:8080",
+		},
+		cli.BoolTFlag{
```
Comment (Oats87) [inline main.go]: Shouldn't these be something like `enable-controller` to be more descriptive, as it's a flag not a "command"?

**Review by Oats87 — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (luthermonson) [reply main.go]: sure, good idea

File: main.go [hunk 3]
```diff
@@ -40,6 +34,19 @@ func main() {
...
+			Name:   "controller",
+			EnvVar: "CONTROLLER",
+		},
```
Comment (dramich) [inline main.go]: Add descriptions for the flags

File: main.go [hunk 4]
```diff
@@ -64,15 +71,25 @@ func main() {
...
 		logrus.SetLevel(logrus.DebugLevel)
 		logrus.SetReportCaller(true)
 	}
 
-	logrus.Info("Starting Terraform Controller")
-	kubeconfig, err := resolvehome.Resolve(c.String("kubeconfig"))
+	ctx := signals.SetupSignalHandler(context.Background())
 
+	go startController(ctx, c)
+	go startApi(ctx, c)
```
Comment (dramich) [inline main.go]: I think doing the flag check here would be cleaner, also makes the intention clear as to what we are running vs having to check the function call to see the flag check

File: pkg/api/api.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rancher/terraform-controller/pkg/api/routes"
+	"github.com/sirupsen/logrus"
+)
+
+func Start(ctx context.Context, address string) error {
+	logrus.Infof("Starting API Server\n")
```
Comment (dramich) [inline pkg/api/api.go]:
```suggestion
	logrus.Info("Starting API Server")
```

File: main.go (hunk 4 above)
Comment (luthermonson) [reply main.go]: sounds good

**Pull request merged**